src/storage/data_table.cpp

```diff
@@ -530,6 +530,72 @@ void DataTable::VerifyNewConstraint(ClientContext &context, DataTable &parent, c
 	local_storage.VerifyNewConstraint(parent, *constraint);
 }
 
+bool HasUniqueIndexes(TableIndexList &list) {
+	bool has_unique_index = false;
+	list.Scan([&](Index &index) {
+		if (index.IsUnique()) {
+			return has_unique_index = true;
+			return true;
```

Comment:
Completely unrelated to this PR but this should probably be an enum not a bool in the API - the code is hard to understand (what do `true` and `false` mean). I might clean that up later.